File: projects/m1/016-next-day/solutions/PiervitoRinaldi/next-day.py
day = int(input('Enter day: '))
month = int(input('Enter month: '))
year = int(input('Enter year: '))


if year % 400 == 0 or year % 4 == 0:

    if month == 12 and day == 31:
        print('1 - 1 -', year +1)
   
    elif month == 1 or month == 3 or month == 5 or month == 7 or month == 8 or month == 10 or month == 12:
        if day >= 1 and day < 31:
            print(day +1, '-', month, '-', year)
        elif day == 31:
            print('1 -', month + 1, '-', year)
    
    elif month == 2:
        if day >= 1 and day < 29:
            print(day +1, '-', month, '-', year)
        elif day == 29:
            print('1 -', month + 1, '-', year)
    
    elif month == 4 or month == 6 or month == 9:
        if day >= 1 and day < 30:
            print(day +1, '-', month, '-', year)
        elif day == 31:
            print('1 -', month + 1, '-', year)

elif year % 400 != 0 or year % 4 != 0:

    if month == 12 and day == 31:
        print('1 - 1 -', year +1)
   
    elif month == 1 or month == 3 or month == 5 or month == 7 or month == 8 or month == 10 or month == 12:
        if day >= 1 and day < 31:
            print(day +1, '-', month, '-', year)
        elif day == 31:
            print('1 -', month + 1, '-', year)
    
    elif month == 2:
        if day >= 1 and day < 28:
            print(day +1, '-', month, '-', year)
        elif day == 28:
            print('1 -', month + 1, '-', year)
    
    elif month == 4 or month == 6 or month == 9:
        if day >= 1 and day < 30:
            print(day +1, '-', month, '-', year)
        elif day == 31:
            print('1 -', month + 1, '-', year)

File: projects/m1/016-next-day/solutions/alessiacaporaso/next-day.js
//date for exemple
const date = new Date();
const currentDate = date.getDate() + "-" + (date.getMonth() + 1) + "-" + date.getFullYear();
// ask the user day, month and year
const choiseDay = prompt('Please, enter the day, month and year', currentDate);
// control the day
const userChoiseDate = choiseDay.split("-");
// divide the array with the date
const day = userChoiseDate[0];
const month = userChoiseDate[1];
const year = userChoiseDate[2];
// control the leap year and number of day
let numberDay;
switch (month) {
  case '4':
  case '6':
  case '9':
  case '11':
    numberDay = 30;
    break;
  case '2':
    if (year % 400 === 0 || year % 4 === 0 && currentYear % 100 !== 0) {
      numberDay = 29
    } else {
      numberDay = 28
    }
    break;
  default:
    numberDay = 31;
    break;
}
// control day, month and year
let newDay;
let newMonth;
let newYear;
if (day < numberDay) {
  newDay = +day + 1;
  newMonth = month;
  newYear = year;
} else if (day >= numberDay && month < 12) {
  newDay = 1;
  newMonth = +month + 1;
  newYear = year
} else {
  newDay = 1;
  newMonth = 1;
  newYear = +year + 1;
}
//display the date after the date choice of the user
alert(`The new date is ${newDay}-${newMonth}-${newYear}`)